src/main/java/narcibot/Narcibot.java

```diff
@@ -0,0 +1,240 @@
+        PrintStream ps = new PrintStream(baos);
+        PrintStream old = System.out;
+        System.setOut(ps);
+        command(parser.parse(input));
+        System.out.flush();
+        System.setOut(old);
+        return baos.toString();
+    }
+
+    /**
+     * Runs the Narcibot program.
+     */
+    public void run()  {
+        System.out.println(ui.welcome());
+        String input;
+        Scanner sc = new Scanner(System.in);
+        while (true) {
+            input = sc.nextLine();
+            if(command(parser.parse(input))) {
+                break;
+            }
+        }
+
+        try {
+                taskList.store(storage.store());
+        } catch (IOException e) {
+            System.out.println("File not found");
+        }
+    }
+
+    private boolean command(String[] command) {
+        boolean end = false;
+        try {
+            switch (command[0]) {
+            case "bye":
+                if (command.length != 1) {
+                    throw new IncorrectFormatException("Please enter only one word for this command");
+                }
+                ui.bye();
+                return true;
+            case "list":
+                if (command.length != 1) {
+                    throw new IncorrectFormatException("Please enter only one word for this command.");
+                }
+                ui.list();
+                taskList.list();
+                break;
+            case "mark":
+                if (command.length != 2) {
+                    throw new IncorrectFormatException("Please enter mark followed by a number for this command. Example: mark 8");
+                }
+                ui.mark();
+                taskList.mark(command[1]);
+                break;
+            case "unmark":
+                if (command.length != 2) {
+                    throw new IncorrectFormatException("Please enter unmark followed by a number for this command. Example: unmark 7");
+                }
+                ui.unmark();
+                taskList.unmark(command[1]);
+                break;
+            case "delete":
+                if (command.length != 2) {
+                    throw new IncorrectFormatException("Please enter delete followed by a number for this command. Example: delete 7");
+                }
+                ui.delete();
+                taskList.delete(command[1]);
+                break;
+            case "todo":
+                if (command.length != 2) {
+                    throw new IncorrectFormatException("You want me to remind you of something but you won't tell me of what it is?");
+                }
+                ui.task(taskList.todo(command[1]));
+                break;
+            case "deadline":
+                if (command.length != 3) {
+                    throw new IncorrectFormatException("You want me to remind you of something but you won't tell me of what it is? The format is deadline (task) /by (time)");
```

Comment:
Shouldn't line length be no longer than 120 characters? 🤔  I have noticed the same issue in several other places too.